packages/test/test-end-to-end-tests/src/test/summarizeRefreshLatestSummary.spec.ts

```diff
@@ -53,5 +48,79 @@ describeCompat(
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+			],
+			async () => {
+				const summaryConfig: ISummaryConfiguration = {
+					...DefaultSummaryConfiguration,
+					...{
+						maxTime: 5000 * 12,
+						maxAckWaitTime: 120000,
+						maxOps: 1,
+						initialSummarizerDelayMs: 0,
+					},
+				};
+				const testContainerConfig: ITestContainerConfig = {
+					runtimeOptions: {
+						summaryOptions: {
+							summaryConfigOverrides: summaryConfig,
+						},
+					},
+					loaderProps: {
+						configProvider: mockConfigProvider({
+							"Fluid.Summarizer.immediatelyRefreshLatestSummaryAck": true,
+						}),
+					},
+				};
+				const container = await provider.makeTestContainer(testContainerConfig);
+				const container2 = await provider.loadTestContainer(testContainerConfig);
+				const dataObject = (await container.getEntryPoint()) as ITestDataObject;
+				const dataObject2 = (await container2.getEntryPoint()) as ITestDataObject;
+				await waitForContainerConnection(container);
+				await waitForContainerConnection(container2);
+
+				// The first summary needs to be generated otherwise the summarizer will only generate one summary.
+				dataObject._root.set(`first`, "op");
+				await provider.ensureSynchronized();
+
+				const array: string[] = [];
+				container.on("op", (op) => {
+					if (op.type === MessageType.SummaryAck) {
+						array.push((op as ISummaryAckMessage).contents.handle);
+					}
+					if (op.type === MessageType.SummaryNack) {
+						throw new Error("Unexpected summary nack");
+					}
+				});
+
+				// Sending ops from two different clients/containers causes the heuristics summarizer to handle ops twice, which causes the summarizer to summarize before acking its first summary.
+				// Sending ops from the same client does not cause the heuristics summarizer to handle ops twice probably because of batching.
+				dataObject._root.set(`a`, "op1");
+				dataObject2._root.set(`b`, "op2");
+				await provider.ensureSynchronized();
+				const container3 = await provider.loadTestContainer(testContainerConfig, {
+					[LoaderHeader.version]: array[0],
+				});
+
+				const dataObject3 = (await container3.getEntryPoint()) as ITestDataObject;
+				assert(dataObject3._root.get(`a`) === "op1");
```

Comment:
Why is this validation needed?

Reply:
It's not

Reply:
This helps test eventual consistency. There's actually some interesting behavior here that may be worth investigating as this is an edge case.